Note.cpp

```diff
@@ -0,0 +1,168 @@
+/******************************************************************
+    Note.cpp
+
+    implementation of the Note structure
+******************************************************************/
+
+#include "Note.h"
+
+Note::Note()
+{
+    stringNum = 1;
+    fretNum = 0;
+    finger = none;
+    //std::cout << "Note default CTOR" << std::endl; 
+}
+
+Note::Note(int str, int fret, Finger digit, char note)
+{
+    stringNum = str;
+    fretNum = fret;
+    finger = digit;
+
+    key = note;
+
+    //std::cout << "Note CTOR" << std::endl;
+}
+
+Note::~Note()
+{
+    //std::cout << "Note DTOR" << std::endl;
+}
+
+int Note::getStringNum()
+{
+    return stringNum;
+}
+
+void Note::setStringNum(int strNum)
+{
+    stringNum = strNum;
+}
+
+int Note::getFretNum()
+{
+    return fretNum;
+}
+
+void Note::setFretNum(int fret)
+{
+    fretNum = fret;
+}
+
+Finger Note::getFinger()
+{
+    return finger;
+}
+
+void Note::setFinger(Finger digit)
+{
+    finger = digit;
+}
+
+void Note::setKey(char k)
+{
+    key = k;
+}
+
+char Note::getKey()
+{
+    return key;
+}
+
+char Note::getNextTone()
+{
+    switch(key)
+    {
+        case 'A':
+            return 'B';
+        case 'B':
+            return 'C';
+        case 'C':
+            return 'D';
+        case 'D':
+            return 'E';
+        case 'E':
+            return 'F';
+        case 'F':
+            return 'G';
+        case 'G':
+            return 'A';
+        default:
+            return 'A';
+    }
+}
+
+char Note::getKeyByInternal(int interval)
+{
+    char semitone = key;
+
+    if(interval == 0)
+    {
+        return semitone;
+    }
```

Comment:
You don't need `semitone` yet.  you don't need it until the `while()` loop.  just return `key` directly here.
- [ ] revise

Reply:
fixed.